### PR TITLE
add dummy value for client-id

### PR DESCRIPTION
**Pull request by timothybest-usds**
## Related Issue or Background Info

our demo env is currently down because `OKTA_OAUTH2_CLIENT_ID` is not defined. This also effects running the test locally because by default no value for `OKTA_OAUTH2_CLIENT_ID` is provided and `client-id` is not overwritten

demo is failing to start and throwing the following exception
```
org.springframework.boot.context.properties.bind.BindException: Failed to bind properties under 'spring.security.oauth2.resourceserver.opaquetoken.client-id' to java.lang.String
```

### CODE REVIEW

**Comment by benwarfield-usds** [comment]
This error is telling us that somebody broke the configuration for demo and local tests--I would prefer that we fix that rather than gag the canary.

**Review by emmanuel-apau — APPROVED** [review]

**Comment by jeremyzitomer-usds** [comment]
+1 to what Ben said, or at least I'd like to understand this better. I've rolled out to `demo` with this configuration setting before without breaking the environment. For what release did this problem start? Is someone using a configuration in the okta.oauth2 property family in `demo` that they weren't using before?

**Comment by timothybest-usds** [comment]
Not going to merge this unless there is an urgent need to update `demo`. @jeremyzitomer-usds is working on a proper fix. I will close this PR once that lands

**Comment by TomNUSDS** [comment]
I like using this approach of supplying a default value. For some of our configurations, having missing data is fine in Dev but NOT in production. Having a `MISSING` value allows the code to log a warning/error that it is not running versus just dying. It would be nice to have a global `IsInProd()` so it's easy to `throw` for missing critical settings just when they are absolutely required.

**Comment by benwarfield-usds** [comment]
I prefer failing fast to finding things in the error log later—too many times I've found something in an error log only way after I needed to know it.

**Pull request closed without merging**
